server/src/main/java/com/cloud/vm/UserVmManagerImpl.java

```diff
@@ -3805,11 +3805,18 @@ public UserVm createAdvancedVirtualMachine(DataCenter zone, ServiceOffering serv
     @ActionEvent(eventType = EventTypes.EVENT_VM_CREATE, eventDescription = "deploying Vm")
     public UserVm finalizeCreateVirtualMachine(long vmId) {
         s_logger.info("Loading UserVm " + vmId + " from DB");
-        UserVm userVm = getUserVm(vmId);
+        UserVmVO userVm = _vmDao.findById(vmId);
         if (userVm == null) {
             s_logger.info("Loaded UserVm " + vmId + " (" + userVm.getUuid() + ") from DB");
         } else {
             s_logger.warn("UserVm " + vmId + " does not exist in DB");
+            VMTemplateVO template = _templateDao.findById(userVm.getTemplateId());
+            if (template != null && template.isEnablePassword()) {
+                String password = _mgr.generateRandomPassword();
+                _vmDao.loadDetails(userVm);
+                userVm.setPassword(password);
+                encryptAndStorePassword(userVm, password);
+            }
```

Comment:
this probably happens somewhere else as well. make a method?

Reply:
@DaanHoogland generate and store method are used differently in different place. A refactoring to consolidate can be done in a separate PR